gef-ui/api-tests/amendments/cover-end-date.api-test.ts

```diff
@@ -0,0 +1,160 @@
+import { Headers } from 'node-mocks-http';
+import { NextFunction, Request, Response } from 'express';
+import { DEAL_STATUS, DEAL_SUBMISSION_TYPE, ROLES } from '@ukef/dtfs2-common';
+import { HttpStatusCode } from 'axios';
+import { withRoleValidationApiTests } from '../common-tests/role-validation-api-tests';
+import app from '../../server/createApp';
+import { createApi } from '../create-api';
+import api from '../../server/services/api';
+import * as storage from '../test-helpers/storage/storage';
+import { Deal } from '../../server/types/deal';
+import { Facility } from '../../server/types/facility';
+
+const originalEnv = { ...process.env };
+
+const { get } = createApi(app);
+
+jest.mock('csurf', () => () => (_req: Request, _res: Response, next: NextFunction) => next());
+jest.mock('../../server/middleware/csrf', () => ({
+  csrfToken: () => (_req: Request, _res: Response, next: NextFunction) => next(),
+}));
+
+const mockGetFacility = jest.fn();
+const mockGetApplication = jest.fn();
+
+const dealId = '123';
+const facilityId = '111';
+const amendmentId = '111';
+
+const mockDeal = { exporter: { companyName: 'test exporter' }, submissionType: DEAL_SUBMISSION_TYPE.AIN, status: DEAL_STATUS.UKEF_ACKNOWLEDGED } as Deal;
```

Comment:
could we use `MOCK_BASIC_DEAL` from './mocks/mock-applications' instead  and then change the fields we need to change?

eg 
`const mockDeal  = {
   ...MOCK_BASIC_DEAL,
   submissionType: DEAL_SUBMISSION_TYPE.AIN, 
   status: DEAL_STATUS.UKEF_ACKNOWLEDGED
`